app/javascript/shared/components/useListNavigation.js

```diff
@@ -211,15 +211,29 @@ function getPreviousElement(
 }
 
 /**
- * Gets the first visible element that matches a selector
+ * Checks if the first completely visible element is present that matches a selector and returns if it is available
+ * If that isn't visible, it looks for the partially visible element that is present and returns that
+ * If no elements visible(like the banner case which could cover the entire viewport), we select the first element from the list
  *
  * @param {string} selector - The CSS selector
  *
  * @returns {object} The first visible element
  */
 function getFirstVisibleElement(selector) {
   const elements = document.querySelectorAll(selector);
-  return Array.prototype.find.call(elements, (element) =>
-    isInViewport(element),
+  const completelyVisibleFirstElement = Array.prototype.find.call(
+    elements,
+    (element) => isInViewport({ element }),
   );
+  if (completelyVisibleFirstElement) {
+    return completelyVisibleFirstElement;
+  }
+  const partiallyVisibleFirstElement = Array.prototype.find.call(
+    elements,
+    (element) => isInViewport({ element, allowPartialVisibility: true }),
+  );
+  if (partiallyVisibleFirstElement) {
+    return partiallyVisibleFirstElement;
+  }
+  return elements && elements[0];
```

Comment:
Refactor (blocking): Instead of Array.prototype, use the `find` method on an actual array. This can be accomplished by [spreading](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax) the [NodeList](https://developer.mozilla.org/en-US/docs/Web/API/NodeList) into an array.

```suggestion
  const elements = [...document.querySelectorAll(selector)];
  const completelyVisibleFirstElement = elements.find((element) =>
    isInViewport({ element }),
  );

  if (completelyVisibleFirstElement) {
    return completelyVisibleFirstElement;
  }

  const partiallyVisibleFirstElement = elements.find((element) =>
    isInViewport({ element, allowPartialVisibility: true }),
  );

  if (partiallyVisibleFirstElement) {
    return partiallyVisibleFirstElement;
  }

  return elements[0];
}
```

Reply:
Sure @nickytonline. Thanks for taking the time to go over. Will push the suggested change.

Reply:
Have pushed it @nickytonline. Kindly check and let me know when you find time :)